### PR TITLE
core: Add `clone` to `Region` class 

### PR DESCRIPTION
**Pull request by ShaolunWang**
#610
- Adding `clone` to `Region` class
- Adding respective test

### CODE REVIEW

File: xdsl/ir/core.py [hunk 1]
```diff
@@ -1668,6 +1668,14 @@ def erase_block(self, block: int | Block, safe_erase: bool = True) -> None:
         block = self.detach_block(block)
         block.erase(safe_erase=safe_erase)
 
+    def clone(self) -> Region:
+        """
+        clone the entire region into a new one.
```
Comment (compor) [inline xdsl/ir/core.py:1673]:
```suggestion
        Clone the entire region into a new one.
```

Docstring comments start with a capital letter.

**Review by compor — COMMENTED** [review]
I don't mind this in, but do we need it?
I know there is an issue up, but maybe we are more interested in keeping the core API small?

**Review by kingiler — APPROVED** [review]
LGTM

**Review by AntonLydike — APPROVED** [review]
:+1:

**Pull request merged**